File: agent-driven-architecture-main/examples/kotlin/block/escalation/src/main/kotlin/adr/blocks/escalation/Port.kt
// ── blocks/escalation/port — the block's PRIVATE frozen contract (4.6/G11) ─
// The paging integration ships INSIDE the block, as port + adapter — never inline in
// a tool and never as a shared service the whole app can reach. An interface only.

package adr.blocks.escalation

import adr.spine.pure.TicketId

interface OncallPort {
    fun page(ticket: TicketId)
}

File: agent-driven-architecture-main/examples/typescript/src/blocks/escalation/port.ts
// ── blocks/escalation/port — the block's PRIVATE frozen contract ───────────
// An interface, nothing else (check C11). The block owns its dependency; the
// composition root binds it. No sibling and no spine file names this type.

import type { TicketId } from "@adr/spine/pure/ids";

export interface OncallPort {
  page(ticket: TicketId): void;
}
